backend/src/main/kotlin/org/loculus/backend/api/FileTypes.kt

```diff
@@ -0,0 +1,28 @@
+/**
+ * Includes API data classes that are used for the file sharing feature.
+ */
+package org.loculus.backend.api
+
+import io.swagger.v3.oas.annotations.media.Schema
+import org.loculus.backend.service.files.FileId
+
+data class FileIdAndUrl(
+    @Schema(description = "The id of the file.", example = "8D8AC610-566D-4EF0-9C22-186B2A5ED793") val fileId: FileId,
+    @Schema(
+        description = "A presigned URL",
```

Comment:
What actions are authorized by the presigned URL? Only read, or also write? Should the authorization information be a property of this class?

Reply:
The kind of URL inside of the data class can be different depending on context. The URL allows either GET or PUT, but not both. For uploading, only PUT is allowed.

> Should the authorization information be a property of this class?

I'm not entirely sure what you mean here, do you mean that the user gets extra info on what the URL can be used for? I'd say that this should be clear from context, i.e. depending on which endpoint you queried, you get a PUT or GET URL.

Reply:
I just thought presigned on its own is vague and differentiating between read vs write in the types might make things clearer in signatures, e.g. knowing that released data contains _read_ urls not _write_ urls

Reply:
Ah I see, thanks for clarifying. So far the type didn't restrict which type of URL to use, but I see how making the name more specific to how it's currently used helps with understanding the data flow as well, so I've renamed it now. I was hesitating to put this constraint in because there isn't really a need, but it makes sense!

04f8f7888